client/src/components/TweetMessage.js

```diff
@@ -0,0 +1,59 @@
+import React, { useEffect, useState } from 'react';
+import reactStringReplace from 'react-string-replace';
+
+const TweetMessage = ({ tweet }) => {
+  const [links, setLinks] = useState([]);
+
+  useEffect(() => {
+    // find all links inside tweet.text string
+    setLinks(
+      tweet.text.match(
+        /(https?:\/\/(?:www\.|(?!www))[^\s]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/gi
+      )
+    );
+  }, [tweet.text]);
+
+  const renderTextMessage = () => {
+    // replace string "links" with an actual link or image
+    // does not seem like the best way to do this
+    let finalMessage;
+    if (links) {
```

Comment:
This will always evaluate to true even if there are no values in the array.
https://www.sitepoint.com/javascript-truthy-falsy/

Reply:
fixed